src/screens/users/index.js

```diff
@@ -1,78 +1,90 @@
 import React, { PureComponent } from 'react';
-import { View, Text, TouchableOpacity } from '../../components';
-import { StyleSheet } from 'react-native'
+import { StyleSheet, TextInput } from 'react-native';
 import { connect } from 'react-redux';
+import { Text, View, Touchable } from '../../components';
+import * as COLORS from '../../constants/colors';
+import { search } from '../../store/users/actions'
 
 class Users extends PureComponent {
-    constructor() {
-        super();
-    }
-
     openChat = (receivingUser) => {
         this.props.navigation.navigate('Chat', { receivingUser });
     }
 
+    handleChange = (type, value) => {
+        this.setState({ [type]: value });
+    }
+
+    handleSubmit = () => {
+        const { name } = this.state
+        const { onSearch } = this.props
+
+        onSearch(name)
+    }
+
     render() {
         return (
             <View style={styles.container}>
-                {
-                    this.props.users.map(user => (
-                        <View key={user.id} style={styles.userContainer}>
-                            <Text style={styles.name}>{user.name}</Text>
-                            <TouchableOpacity
-                                style={styles.buttonContainer}
-                                onPress={() => this.handleOpenChat(user)}
-                            >
-                                <Text style={styles.buttonText}>Chat</Text>
-                            </TouchableOpacity>
-                        </View>
-                    ))
-                }
+                <TextInput
+                    onChangeText={value => this.handleChange('name', value)}
+                    autoCapitalize='none'
+                    style={styles.input}
+                />
+                <Touchable
+                    onPress={this.handleSubmit}
+                    style={styles.button}
+                >
+                    <Text style={styles.buttonText}>Search</Text>
+                </Touchable>
             </View>
         );
     }
 }
 
 
-export default connect(state => ({
-    users: state.users.length ? state.users.filter(user => user.id !== state.user.id) : []
-}))(Users);
+export default connect(
+    state => ({
+        users: state.users
+    }),
+    dispatch => ({
+        onSearch: (name) => dispatch(search(name)),
+    }))(Users);
 
 const styles = StyleSheet.create({
     container: {
         justifyContent: 'center',
         alignItems: 'center',
-        backgroundColor: 'steelblue',
+        backgroundColor: COLORS.RED,
         height: '100%',
```

Comment:
Default style in React Native is FlexBox. Use 
`flex: 1` instead  ` height: '100%',  width: '100%'`


For some specific styles, this can be needed:
```
import { Dimensions } from "react-native";
const DEVICE_HEIGHT = Dimensions.get("window").height;
const DEVICE_WIDTH = Dimensions.get("window").width;
```

Reply:
I have root container with flex:1. In this case, I can write